packages/main/src/plugin/index.ts

```diff
@@ -2320,6 +2320,47 @@ export class PluginSystem {
       },
     );
 
+    const withTaskConnection = async (options: {
```

Comment:
Can't we have this in a dedicated files ? like something utils?

The `index.ts` is pretty big, and this help reducing its complexity, but would be even better in a dedicated files

Reply:
This is not easy to extract, as it contains many dependencies to local objects (this.getLogHandler, cancellationTokenRegistry, taskManager)

Reply:
I don't know... this feel like this is just making it hard to test, and make the index file more complex again :crying_cat_face: 

I draft something ~ the cancellationTokenRegistry and taskManager are just class, so they can be passed down, the `getLogHandler` can be binded to keep the same behavior, your tests are passing with this

https://github.com/axel7083/podman-desktop/tree/refactor/extract-task-connection-logic

Reply:
I'm not sure I want to externalize this function. It is specialized to these 2 ipcHandles only (3 when createVmProviderConnection will be added), and is not generic enough to be used for other places where there is a similar pattern (the ones introduced in https://github.com/podman-desktop/podman-desktop/pull/8144).

My idea is to have this small refactor limited to these 2/3 ipcHandles, as a step towards refactoring all the ipcHandles using this pattern. But I'm not sure I want to refactor everything now. @benoitf WDYT?

Reply:
on my end, I prefer to have only class/methods, no inline arrow functions

so new TaskConnection(.....).someThing()

I think it's ok to have something that works only for ProviderConnection objects being externalized

Reply:
I've added a class based on Axel's draft